### PR TITLE
Fixes #5050 - Server tests with deviceId login

### LINKED ISSUE #5050: Server tests to support logIn by deviceId
With this change https://github.com/mozilla-services/screenshots/pull/4726/files#diff-e79f0cf7ae229b6047499adeeea99ae6R36, we have our tests in master running with both deviceId and accountId.

Update clientlib.py to allow login only using deviceId and support login using both deviceId and accountId to test set expiration flow

See https://github.com/mozilla-services/screenshots/pull/4966#issuecomment-431083898

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

File: test/server/clientlib.py [hunk 1]
```diff
@@ -227,11 +231,18 @@ def make_random_id():
...
+        if hasAccount:
+            user = ScreenshotsClient(backend=backend, hasAccount=True)
+        else:
+            user = ScreenshotsClient(backend=backend)
     else:
-        user = ScreenshotsClient()
+        if hasAccount:
+            user = ScreenshotsClient(hasAccount=True)
+        else:
+            user = ScreenshotsClient()
```
Comment (punamdahiya) [inline test/server/clientlib.py]: had to break it into if else to make python happy, in ideal world  user = ScreenshotsClient(backend=backend, hasAccount=hasAccount) should work

Comment (ianb) [reply test/server/clientlib.py]: That's weird, `ScreenshotsClient(backend=backend, hasAccount=hasAccount)` should work fine...?

Another option would be:

```py
def screenshots_session(**kw):
    user = ScreenshotsClient(**kw)
```

Comment (punamdahiya) [reply test/server/clientlib.py]: Using kwargs works and simplifies it, updated PR

**Comment by punamdahiya** [comment]
Resolved conflict and rebased with latest master

**Comment by punamdahiya** [comment]
@ianb please confirm if it's ok to carry over your review+ on rebased PR. Thanks

**Pull request merged**